### PR TITLE
LPD-44820 Custom object definitions cannot be bound to modifiable system object definitions

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://liferay.atlassian.net/browse/LPD-44820

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 1]
```diff
@@ -1886,6 +1886,17 @@ private void _validateEdge(
...
+				  !objectDefinition2.isModifiableAndSystem()) ||
+				 (!objectDefinition1.isModifiableAndSystem() &&
+				  objectDefinition2.isModifiableAndSystem())) {
+
+			throw new ObjectRelationshipEdgeException(
+				"Inheritance between modifiable system and custom object " +
+					"definitions is not allowed",
+				"inheritance-between-modifiable-system-and-custom-object-" +
+					"definitions-is-not-allowed");
+		}
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:1899]: I am uncertain whether the existing 'if' statement can be used to simplify my logic or if it is necessary to always sort alphabetically by error message

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 2]
```diff
@@ -203,6 +203,31 @@ public void testAddObjectRelationship() throws Exception {
 				"able", false, ObjectRelationshipConstants.TYPE_MANY_TO_MANY,
 				null));
 
+		AssertUtils.assertFailure(
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:206]: Why is there an empty line between this new assert and the existing one `DuplicateObjectRelationshipException.class`?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 3]
```diff
@@ -1854,6 +1879,23 @@ public void testUpdateObjectRelationship() throws Exception {
...
+					_objectRelationshipLocalService,
+					_modifiableSystemObjectDefinition, _objectDefinition1)));
+		AssertUtils.assertFailure(
+			ObjectRelationshipEdgeException.class,
+			"Inheritance between modifiable system and custom object " +
+				"definitions is not allowed",
+			() -> _bindObjectDefinitions(
+				ObjectRelationshipTestUtil.addObjectRelationship(
+					_objectRelationshipLocalService, _objectDefinition1,
+					_modifiableSystemObjectDefinition)));
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:1897]: Could you please tell me why you're testing adding in the update method?
we can talk in person :+1:

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java (hunk 1 above)
Comment (marcelabc) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:1899]: :thinking:

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 3 above)
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:1897]: I'm unsure if we must keep these asserts closer to the creation of _objectDefinition1.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:206]: No reason. I removed it!

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 3 above)
Comment (carolmariaabb) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:1897]: But `_objectDefinition1 ` is created in the `setUp`  method [here](https://github.com/liferay-bpm/liferay-portal/blob/66fe3a418adb10770f39b636777cc4e3f39f593d/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java#L144-L151)

**Comment by carolmariaabb** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:report:97213773

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**